src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java

```diff
@@ -174,22 +180,45 @@ public Response<UserResponseDTO> getStudentById(String id) {
     }
 
     /**
-     * Retrieves a list of all users from the database.
+     * Retrieves a list of all users from the database by page.
      *
      * @return A response containing a list of user information.
      */
-    public Response<List<UserResponseDTO>> getAllUsers() {
-        Response<List<UserResponseDTO>> response = new Response<>();
-        List<UserDAO> userDAOS = userRepository.findAll();
-        List<UserResponseDTO> usersList = new ArrayList<>();
-        userDAOS.forEach(userDAO -> usersList.add(this.modelMapper.map(userDAO, UserResponseDTO.class)));
-        if (usersList.isEmpty())
-            response.setMeta(Meta.builder().message("DataBase is Empty").statusCode(HttpStatus.NOT_FOUND.value()).build());
-        else
-            response.setMeta(Meta.builder().message("Retrieval Successful").statusCode(HttpStatus.OK.value()).build());
-        log.info("Total Users: {}",usersList.size());
-        log.info("Users Details: {}", usersList);
-        response.setData(usersList);
-        return response;
+    public Response<List<UserResponseDTO>> getAllUsers(int page, int size, String sortBy, String sortOrder) {
+        try {
+            Sort sort = Sort.by(Sort.Direction.fromString(sortOrder), sortBy);
+            Pageable pageable = PageRequest.of(page, size, sort);
+            Page<UserDAO> userPage = userRepository.findAll(pageable);
+
+            if (!userPage.hasContent()) {
+                Response<List<UserResponseDTO>> emptyResponse = new Response<>();
```

Comment:
do we need to create response object twice?

Reply:
Also @Umaid-gh can we use builder from lombok wherever necessary rather than using new keyword